### PR TITLE
Define our own externals for Node.js fs and os APIs

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: okio/src/jsMain/kotlin/okio/fs.kt [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+ * [kotlinx_nodejs]: https://github.com/Kotlin/kotlinx-nodejs
+ */
+@file:JsModule("fs")
+@file:JsNonModule
+package okio
+
+import org.khronos.webgl.Uint8Array
+import kotlin.js.Date
+
+internal external fun closeSync(fd: Number)
```
Comment (swankjesse) [inline okio/src/jsMain/kotlin/okio/fs.kt:79]: `internal external fun` ?
that means we can have fun EVERYWHERE

File: okio/src/jsMain/kotlin/okio/fs.kt [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This class declares the subset of Node.js filesystem APIs that we need in Okio.
```
Comment (Egorand) [inline okio/src/jsMain/kotlin/okio/fs.kt:18]: As long as all of these aren't API we may retire them later if Dukat or kotlinx-nodejs improve

**Review by Egorand — APPROVED** [review]

File: okio/src/jsMain/kotlin/okio/fs.kt (hunk 1 above)
Comment (Egorand) [reply okio/src/jsMain/kotlin/okio/fs.kt:79]: This has to be a tweet

File: okio/src/jsMain/kotlin/okio/fs.kt (hunk 2 above)
Comment (swankjesse) [reply okio/src/jsMain/kotlin/okio/fs.kt:18]: definitely

Comment (swankjesse) [reply okio/src/jsMain/kotlin/okio/fs.kt:18]: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50341/files !

**Pull request merged**